---
title: "Find a gene project - Kcnh2 (a VG potassium channel)"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

First, read in the alignment file
```{r}
aln.kcnh2 <- read.fasta("muscle_aln_kcnh2_rev.fst")
```

Trim down to the original sequence of interest
```{r}
aln.kcnh2$ali <- aln.kcnh2$ali[ , 512:825]
View(aln.kcnh2)
```

Write new fasta file using the trimmed aligned sequences
```{r}
aln.kcnh2.new <- write.fasta(aln.kcnh2, file = "muscle_aln_kcnh2_trim")
```

Create percent identity matrix; first trying seqidentity function
```{r}
aln.kcnh2.mat <- seqidentity(aln.kcnh2)
View(aln.kcnh2.mat)
```

Make a heatmap using the identity matrix:
```{r}
library(pheatmap)
par(mai=c(8, 3, 9, 1))
alnment.heatmap <- pheatmap(aln.kcnh2.mat)

```


Q8: Blast pdb for the similar structures. Using the AN for human KCNH2 because the nemtode sequence has a lot of gaps
```{r}
kcnh2.blast <- blast.pdb("AAI27674.1")

kcnh2.blast.trim <- kcnh2.blast$hit.tbl[1:3, ]

kcnh2.blast.plot <- plot.blast(kcnh2.blast)
```


Annotate the pdb outputs
```{r}
pdb.ids <-kcnh2.blast.trim$subjectids
pdb.annotate(pdb.ids)
kcnh2.anno <- pdb.annotate(pdb.ids, anno.terms = c("structureId", 
                                                   "experimentalTechnique",
                                                   "resolution",
                                                   "source"))
View(kcnh2.anno)
```

change rownames
```{r}
rownames(kcnh2.anno) <- kcnh2.anno$structureId
kcnh2.anno <- kcnh2.anno[, -1]
head(kcnh2.anno)
```

Adding the e-value and identity columns to this annotated df to obtain a table of the top 3 unique hits
```{r}
kcnh2.blast.trim.eval <- kcnh2.blast.trim[, c("identity", "evalue")]
kcnh2.top3.pdb <- cbind(kcnh2.anno, kcnh2.blast.trim.eval)
```

Generate a pdb file of one of the proteins from the previous questions. Then write a new pbd file that can be viewed in vmd
```{r}
kcnh2.str <- read.pdb("5VA1")
write.pdb(kcnh2.str)
```

```{r}
pdb.unique
```

